ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js

```diff
@@ -38,9 +38,7 @@
          }
          //TODO: to support writing in different locales \d should be replaced by [0-9] for different locales
          #matchArray ={
-             "integer":"^[+-]?{digits}*$",
-             "decimal":"^[+-]?{digits}{leading}({decimal}{digits}{fraction})?$",
-             "float":"^[+-]?{digits}*({decimal}{digits}*)?$"
+             "integer":"^[+-]?{digits}*$"
```

Comment:
Why have you removed this check for decimal and float ?

Reply:
In v2 we have 2 number type `Decimal` and `Integer`. For v2 we are using HTML`type='number'`, it is already equivalent to 'decimal`, we just need to validate for `integer` . 
In V1 we have `type='text'`, may be that's why we have those checks.